caravel/assets/javascripts/explorev2/actions/exploreActions.js

```diff
@@ -48,18 +48,14 @@ export function setFilterColumnOpts(filterColumnOpts) {
 }
 
 export function resetFormData() {
-  // Clear all form data when switching datasource
+  // Clear all form data when switching datasource or viz_type
   return { type: RESET_FORM_DATA };
```

Comment:
i'm curious why we need to reset form data when switching a datasource? the way i would see this working is when a datasource changes, it should trigger a fetch of datasource meta data which, once returned will update state with the new meta data. the meta data should include select options and any defaults needed.